deepchecks/core/suite.py

```diff
@@ -115,6 +115,47 @@ def select_results(self, idx: Set[int]) -> List[Union[
                 output.append(result)
         return output
 
+    def select_results_by_check_type(
+        self,
+        check_type: Union[Type[BaseCheck], Sequence[Type[BaseCheck]]]
+    ) -> List['check_types.BaseCheckResult']:
+        """Select results by check type(s).
+
+        Parameters
+        ==========
+        check_type : Union[Type[BaseCheck], Sequence[Type[BaseCheck]]]
+            a check type(s), results of which to return
+
+        Returns
+        =======
+        List['check_types.BaseCheckResult']:
+            list of results and failures
+
+        Examples
+        ========
+        >> from deepchecks.tabular.suites import data_integrity
+        >> from deepchecks.tabular.checks import IsSingleValue
+        >>
+        >> suite_result = data_integrity(...)
+        >> check_results = suite_result.select_results_by_check_type(IsSingleValue)
+        >>
+        """
+        if isinstance(check_type, Sequence):
+            if len(check_type) == 0:
+                raise ValueError('Empty sequence is not allowed')
+            check_type = tuple(check_type)
+        elif isinstance(check_type, type) and issubclass(check_type, BaseCheck):
+            check_type = (check_type,)
+        else:
+            t = type(check_type).__name__
+            raise TypeError(f'Unsupported type of "check_type" parameter - {t}')
+
+        return [
```

Comment:
Would convert into regular for and add warning / exception for when:
1. the check requested is not part of the suite
2. the check requested run failed for some reason (not sure if a check result is created in this case)

Reply:
1. added warning
2. `CheckFailure` instances are created for failed checks and a resulting list will include thems, IMO, it is bad idea to raise an exception for them/